python/cuml/cuml/neighbors/weights.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: Apache-2.0
+
+import cupy as cp
+
+
+def compute_weights(distances, weights):
+    """
+    Compute weights from distances for KNN algorithms.
+
+    Parameters
+    ----------
+    distances : cupy.ndarray or array-like
+        The input distances (n_samples, k).
+
+    weights : {'uniform', 'distance'} or callable
+        The kind of weighting used.
+
+    Returns
+    -------
+    weights_arr : cupy.ndarray
+        Weights of shape (n_samples, k).
+        For uniform weights, returns array of 1/k (normalized).
+        For distance weights, returns raw inverse distances (not normalized).
+        For callable, returns raw result of callable(distances) (not normalized).
+    """
+    # Convert to cupy array if needed and ensure 2D
+    if not isinstance(distances, cp.ndarray):
+        distances = cp.asarray(distances)
+
+    # Ensure distances is 2D
+    if distances.ndim == 1:
+        distances = distances.reshape(-1, 1)
+    elif distances.ndim != 2:
+        raise ValueError(
+            f"distances must be 1D or 2D, got shape {distances.shape}"
+        )
+
+    if weights in (None, "uniform"):
+        # Uniform weights: all neighbors contribute equally
+        n_neighbors = distances.shape[1]
+        return cp.full_like(distances, 1.0 / n_neighbors, dtype=cp.float32)
```

Comment:
Is `compute_weights` ever called with `weights="uniform"` or `weights=None`? From my reading of the code I don't think that's possible.